### PR TITLE
Add new common NXP dockerfile

### PR DESCRIPTION
**Pull request by Martin-NXP**
rw61x and k32w1 will share the same new SDK, the objective of this PR is to prepare a common docker image to be able to add new SDK support in next PRs.

### CODE REVIEW

File: integrations/docker/images/stage-2/chip-build-nxp/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    && : # last line
+
+WORKDIR /opt/nxp/
+
+RUN set -x \
+    && git clone --branch v1.4.0-pvw1 https://github.com/NXP/nxp_matter_support.git \
+    && pip3 install --break-system-packages -U --no-cache-dir west \
+    && ./nxp_matter_support/scripts/update_nxp_sdk.py --platform common \
+    && : # last line
+
```
Comment (marius-alex-tache) [inline integrations/docker/images/stage-2/chip-build-nxp/Dockerfile:18]: Should we add a `COPY --from=build <> /opt/nxp` here to update the VSCode docker image easier? E.g. https://github.com/project-chip/connectedhomeip/blob/master/integrations/docker/images/vscode/chip-build-vscode/Dockerfile#L50

Comment (andy31415) [reply integrations/docker/images/stage-2/chip-build-nxp/Dockerfile:18]: we should make sure the vscode dockerfile can still compile NXP apps. Currently we seem to copy /opt/sdk to /opt/k32w  however I am unsure if this is correct anymore. The bits from the new docker file have to be copied over too. The intent is to allow maintainers to only use one image to compile anything.

https://github.com/project-chip/connectedhomeip/blob/master/integrations/docker/images/vscode/chip-build-vscode/Dockerfile#L50

File: integrations/docker/images/stage-2/chip-build-nxp/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ENV NXP_UPDATE_SDK_SCRIPT_DOCKER=/opt/nxp/nxp_matter_support/scripts/update_nxp_sdk.py
```
[the remark this answers is not in the document]
Comment (Martin-NXP) [reply integrations/docker/images/stage-2/chip-build-nxp/Dockerfile:19]: Hello,
When this environment variable is set, we will import dynamically this python file from the nxp.py script to get the path of supported platforms SDKs

File: integrations/docker/images/stage-2/chip-build-nxp/Dockerfile (hunk 1 above)
Comment (marius-alex-tache) [reply integrations/docker/images/stage-2/chip-build-nxp/Dockerfile:18]: The `COPY --from=k32w /opt/sdk /opt/k32w` will still be valid. We need to add here a copy statement that copies the common sdk to `/opt/nxp` for example, then update the vscode Dockerfile to add `COPY --from=nxp /opt/nxp /opt/nxp`. Please note that RW will still use an older Docker image for now, so we still need to keep `COPY --from=rw61x /opt/sdk /opt/nxp-sdk`.

Comment (Martin-NXP) [reply integrations/docker/images/stage-2/chip-build-nxp/Dockerfile:18]: Thank you Andy and Marius for your comments, I pushed the code to fix the missing COPY

**Review by andy31415 — APPROVED** [review]

**Pull request merged**